stock_inventory_discrepancy/models/stock_quant.py

```diff
@@ -90,3 +90,39 @@ def action_apply_inventory(self):
             )
             return action
         return super().action_apply_inventory()
+
+    def user_has_groups(self, groups):
+        # - Allow specific group to validate inventory
+        # - Allow validate on pending status
+        if (
+            self.env.context.get("from_apply_inventory")
+            and groups == "stock.group_stock_manager"
+            and not self.user_has_groups("stock.group_stock_manager")
+            and not self.user_has_groups(
+                "stock_inventory_discrepancy.group_stock_inventory_validation"
+            )
+            and not self.user_has_groups(
+                "stock_inventory_discrepancy.group_stock_inventory_validation_always"
+            )
+        ):
+            raise UserError(
+                _("Only a stock manager can validate an inventory adjustment.")
+            )
+        return super().user_has_groups(groups)
+
+    def _apply_inventory(self):
+        if (
+            not self.user_has_groups("stock.group_stock_manager")
+            and not self.user_has_groups(
+                "stock_inventory_discrepancy.group_stock_inventory_validation"
+            )
+            and not self.user_has_groups(
+                "stock_inventory_discrepancy.group_stock_inventory_validation_always"
+            )
+        ):
+            raise UserError(
+                _("Only a stock manager can validate an inventory adjustment.")
+            )
+        # Allow to write last_inventory_date on stock.location
+        self = self.sudo().with_context(from_apply_inventory=True)
+        return super(StockQuant, self)._apply_inventory()
```

Comment:
this should be super()._apply_inventory()

Reply:
Would that use the new self?

Reply:
ah sorry, I didn't pay attention to this, you are right indeed.

Reply:
I thought `super()._apply_inventory()` would be valid. It should use the new self if I'm not mistaken.